src/chttpd/src/chttpd_httpd_handlers.erl

```diff
@@ -233,11 +233,21 @@ handler_info('POST', [Db, <<"_compact">>], _) ->
 handler_info('GET', [Db, <<"_design">>, Name], _) ->
     {'db.design.doc.read', #{'db.name' => Db, 'design.id' => Name}};
 
-handler_info('POST', [Db, <<"_design">>, Name], _) ->
-    {'db.design.doc.write', #{'db.name' => Db, 'design.id' => Name}};
+handler_info('POST', [Db, <<"_design">>, Name], Req) ->
+    case chttpd:qs_value(Req, "rev") of
+        undefined ->
+            {'db.design.doc.write', #{'db.name' => Db, 'design.id' => Name}};
+        _ ->
+            {'db.design.doc.update', #{'db.name' => Db, 'design.id' => Name}}
+    end;
 
-handler_info('PUT', [Db, <<"_design">>, Name], _) ->
-    {'db.design.doc.write', #{'db.name' => Db, 'design.id' => Name}};
+handler_info('PUT', [Db, <<"_design">>, Name], Req) ->
+    case chttpd:qs_value(Req, "rev") of
```

Comment:
Should these (ddoc PUT and POST) be updated as well with `has_rev`?

Reply:
missed that